### PR TITLE
Decoupling: Move `reshapeStoryObject` to `wp-dashboard`

### PR DESCRIPTION
**Pull request by mohdsayed**
## Context

https://github.com/google/web-stories-wp/pull/9237#discussion_r721221647

## Summary

Move `reshapeStoryObject` utility function from `dashboard` to `wp-dashboard`.

## To-do

- [x] Move reshapeStoryObject to wp-dashboard
- [x] Fix tests
- [x] Fix storybook with latest config changes

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Partially address: https://github.com/google/web-stories-wp/issues/2918

### CODE REVIEW

File: packages/wp-dashboard/src/api/utils/reshapeStoryObject.js [hunk 1]
```diff
@@ -15,9 +15,9 @@
  */
 
 /**
- * Internal dependencies
+ * External dependencies
  */
-import { REST_LINKS } from '../../constants';
+import { REST_LINKS } from '@web-stories-wp/dashboard';
```
Comment (swissspidy) [inline packages/wp-dashboard/src/api/utils/reshapeStoryObject.js]: Can we move `REST_LINKS` to `wp-dashboard` instead?

File: packages/dashboard/src/dataUtils/formattedStoriesArray.js [hunk 2]
```diff
@@ -147,12 +143,12 @@ const formattedStoriesArray = [
             href: 'http://localhost:8899/wp-json/web-stories/v1/web-story/167',
           },
         ],
-        [REST_LINKS.EDIT]: [
+        'wp:action-edit': [
```
Comment (spacedmonkey) [inline packages/dashboard/src/dataUtils/formattedStoriesArray.js:146]: Why change this?

Comment (mohdsayed) [reply packages/dashboard/src/dataUtils/formattedStoriesArray.js:146]: For moving `REST_LINK` to wp-dasbhoard for now. https://github.com/google/web-stories-wp/pull/9345#discussion_r725221772

All `wp:*` will also be removed from dashboard package in upcoming PRs.

File: packages/wp-dashboard/src/api/utils/test/reshapeStoryObject.js [hunk 3]
```diff
@@ -14,11 +14,18 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+/**
+ * Internal dependencies
+ */
```
Comment (swissspidy) [inline packages/wp-dashboard/src/api/utils/test/reshapeStoryObject.js]: ?

File: packages/wp-dashboard/src/api/utils/reshapeStoryObject.js [hunk 4]
```diff
@@ -14,10 +14,13 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
```
Comment (swissspidy) [inline packages/wp-dashboard/src/api/utils/reshapeStoryObject.js]: ?

**Review by swissspidy — APPROVED** [review]

**Pull request merged**